src/matlab2tikz.m

```diff
@@ -349,21 +341,44 @@ function matlab2tikz(varargin)
              clickableUrl(m2t.about.wiki, 'wiki'));
 
     %% Save the figure as TikZ to file
-    saveToFile(m2t, fid, fileWasOpen);
+    m2t = saveToFile(m2t, fid, fileWasOpen);
 
     %% Check for a new matlab2tikz version outside version control
-    if m2t.cmdOpts.Results.checkForUpdates
-        m2tUpdater(m2t.about, m2t.cmdOpts.Results.showInfo);
+    if m2t.args.checkForUpdates
+        m2tUpdater(m2t.about, m2t.args.showInfo);
     end
 
 end
 % ==============================================================================
+function [m2t, counterValue] = getNextValue(m2t, counterName)
```

Comment:
I think getNextValue should better be named increaseValue or increaseGlobalValue.

Also how much work would it be to remove counterValue and just refert to m2t.count.* in the code?